src/main/java/com/google/sps/servlets/getBlobKeyServlet.java

```diff
@@ -46,17 +46,21 @@ public getBlobKeyServlet(){
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
  
-    // get user id -- get entity from user id -- new func!!
+    // get the user entity
     Cookie cookies[] = request.getCookies();
+    if(cookies == null){
+        response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
```

Comment:
Nice job. You should also do this if `userEntity` is null.

Reply:
Okay, done.